components/google_drive/sources/new-or-modified-files-polling/new-or-modified-files-polling.mjs

```diff
@@ -0,0 +1,220 @@
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+import googleDrive from "../../google_drive.app.mjs";
+import { getListFilesOpts } from "../../common/utils.mjs";
+import sampleEmit from "./test-event.mjs";
+import { GOOGLE_DRIVE_FOLDER_MIME_TYPE } from "../../common/constants.mjs";
+
+export default {
+  key: "google_drive-new-or-modified-files-polling",
+  name: "New or Modified Files (Polling)",
+  description: "Emit new event when a file in the selected Drive is created, modified or trashed. [See the documentation](https://developers.google.com/drive/api/v3/reference/changes/list)",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    googleDrive,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+    drive: {
+      propDefinition: [
+        googleDrive,
+        "watchedDrive",
+      ],
+      description: "Defaults to My Drive. To select a [Shared Drive](https://support.google.com/a/users/answer/9310351) instead, select it from this list.",
+      optional: false,
+    },
+    files: {
+      type: "string[]",
+      label: "Files",
+      description: "The specific file(s) you want to watch for changes. Leave blank to watch all files in the Drive. Note that events will only be emitted for files directly in these folders (not in their subfolders, unless also selected here)",
+      optional: true,
+      default: [],
+      options({ prevContext }) {
+        const { nextPageToken } = prevContext;
+        return this.googleDrive.listFilesOptions(nextPageToken, this.getListFilesOpts());
+      },
+    },
+    folders: {
+      type: "string[]",
+      label: "Folders",
+      description: "The specific folder(s) to watch for changes. Leave blank to watch all folders in the Drive.",
+      optional: true,
+      default: [],
+      options({ prevContext }) {
```

Comment:
Is there any reason not to use the `fileId` & `folderId` propDefinitions from the app file?